### ISSUE TITLE
Function argument parameters

### ISSUE THREAD

**Issue by b0mbie** [luttje/glua-api-snippets#37]
Functions like [`net.Receive`](https://wiki.facepunch.com/gmod/net.Receive) and [`concommand.Add`](https://wiki.facepunch.com/gmod/concommand.Add) have function parameters defined in function arguments' descriptions.

The arguments are defined in a bulleted list. Example for `callback` in `concommand.Add` in markup:
```xml
The function to run when the concommand is executed. Arguments passed are:
* <page>Player</page> ply - The player that ran the concommand. NULL entity if command was entered with the dedicated server console.
* <page>string</page> cmd - The concommand string (if one callback is used for several concommands).
* <page>table</page> args - A table of all string arguments.
* <page>string</page> argStr - The arguments as a string.
```

This, by hand, could be written as:
```lua
--- @param callback fun(ply: Player, cmd: string, args: table, argStr: string) # The function to run when the concommand is executed.
```

One possible problem with implementing this in the scraper is that the last sentence, `Arguments passed are:`, is meaningless if included in the argument description, thus it would have to be excluded. So, I think that functions with defined parameters can be hard-coded.

**Comment by luttje** [comment]
Good catch!

I think in this case it would be easiest to just hard-code these overrides in the [`custom/`](https://github.com/luttje/glua-api-snippets/tree/main/custom) directory.

**Comment by b0mbie** [comment]
Yep, from looking at the scraper code, this seems to be right.
I'll see what I can do. It's been awesome using the definitions provided by the addon already, and I think having function arguments with defined parameters would be very useful.

**Comment by b0mbie** [comment]
I've implemented a big part of this in #40.
